controlplane/kubeadm/internal/controllers/helpers.go

```diff
@@ -307,8 +307,9 @@ func (r *KubeadmControlPlaneReconciler) generateMachine(ctx context.Context, kcp
 			Bootstrap: clusterv1.Bootstrap{
 				ConfigRef: bootstrapRef,
 			},
-			FailureDomain:    failureDomain,
-			NodeDrainTimeout: kcp.Spec.MachineTemplate.NodeDrainTimeout,
+			FailureDomain:           failureDomain,
+			NodeDrainTimeout:        kcp.Spec.MachineTemplate.NodeDrainTimeout,
+			NodeVolumeDetachTimeout: kcp.Spec.MachineTemplate.NodeVolumeDetachTimeout,
```

Comment:
@snehala27 Thx for the PR. While we're already here. Can you please also set NodeDeletionTimeout here and drop the if in l.315? Thx!

Reply:
It would be also great if we can add test coverage for all 3 fields in `TestKubeadmControlPlaneReconciler_generateMachine`

Reply:
Sure